File: beecrowd/dart-2.9/data-structures-and-libraries/3160-friends.dart
import 'dart:convert';
import 'dart:io';

void main() {
  final l1 = _getStringListLine();
  final l2 = _getStringListLine();

  final target = _getLine();

  final index = l1.indexOf(target);

  if (target == 'nao') {
    l1.addAll(l2);
  } else {
    l1.insertAll(index, l2);
  }

  print(l1.map((name) => name).toList().join(' '));
}

String _getLine() =>
    stdin.readLineSync(encoding: Encoding.getByName('utf-8'))?.trim() ?? '';

List<String> _getStringListLine() => _getLine().split(' ');

File: beecrowd/javascript/data-structures-and-libraries/3160-friends.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

const l1 = lines.shift().split(' ');
const l2 = lines.shift().split(' ');

const target = lines.shift();

const index = l1.indexOf(target);

if (target == 'nao') {
  l1.push(...l2);
} else {
  l1.splice(index, 0, ...l2);
}

console.log(l1.join(' '));
